eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,110 @@
+<Project>
+  <PropertyGroup>
+    <LinkerTestDir>$(ArtifactsBinDir)linkerTests/</LinkerTestDir>
+    <LinkerTestProjectsDir>$(LinkerTestDir)projects/</LinkerTestProjectsDir>
+    <TestDotNetPath>$(DotNetRoot)dotnet</TestDotNetPath>
+    <ProjectTemplate>$(MSBuildThisFileDirectory)project.csproj.template</ProjectTemplate>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="$(MSBuildProjectDirectory)/*.cs" />
+
+    <TestSupportFiles Include="$(MSBuildThisFileDirectory)SupportFiles/Directory.Build.props">
+      <DestinationFolder>$(LinkerTestDir)</DestinationFolder>
+    </TestSupportFiles>
+    <TestSupportFiles Include="$(MSBuildThisFileDirectory)SupportFiles/Directory.Build.targets">
+      <DestinationFolder>$(LinkerTestDir)</DestinationFolder>
+    </TestSupportFiles>
+  </ItemGroup>
+
+  <Target Name="CreateTestDir"
+          Inputs="@(TestSupportFiles)"
+          Outputs="@(TestSupportFiles->'%(DestinationFolder)\%(FileName)%(Extension)')">
+    <MakeDir Directories="%(TestSupportFiles.DestinationFolder)" />
+    <Copy SourceFiles="@(TestSupportFiles)" DestinationFolder="%(TestSupportFiles.DestinationFolder)" />
+  </Target>
+
+  <Target Name="GetTestConsoleApps">
+    <ItemGroup>
+      <TestConsoleAppSourceFiles>
+        <ProjectDir>$(LinkerTestProjectsDir)$(MSBuildProjectName)/%(Filename)/$(PackageRID)/</ProjectDir>
+        <TestRuntimeIdentifier>$(PackageRID)</TestRuntimeIdentifier>
+      </TestConsoleAppSourceFiles>
+      <TestConsoleAppSourceFiles>
```

Comment:
Why did you split these in two separate Item declarations?

Reply:
Because if you want to use a computed metadata as part of another item metadata, the first one have to be calculated already. So in this case, I am using ProjectDir and TestRuntimeIdentifier on the rest of the metadata below, and if I don't separate then ProjectDir would not be defined yet so I wouldn't be able to do this.

Reply:
I see; maybe worth a comment? So that if someone touches this doesn't go aggressive on cleaning this up and spending time figuring out why it doesn't work?